src/irr/asset/CImageLoaderOpenEXR.cpp

```diff
@@ -29,12 +29,65 @@ namespace irr
 			if (!_file)
 				return {};
 
+			SContext ctx;
+			const auto& fileName = _file->getFileName().c_str();
+
+			if (!readMagicNumber())
+				return {};
+
+			readVersionField();
+			readHeader(fileName, ctx);
```

Comment:
would be cool to re-use this for checking if file can be loaded.

Reply:
> for checking if file can be loaded.
that will be checked by `magicNumber`, sure